torch/_inductor/config.py

```diff
@@ -28,8 +28,13 @@
 benchmark_harness = True
 
 # control store vs recompute heuristic
-realize_reads_threshold = 4
+# For fanouts, rematearialization can lead to exponential blowup. So, have
+# smaller threashold
+realize_reads_remat_threshold = 4
+
 realize_bytes_threshold = 2000
+realize_reads_threshold = 8
```

Comment:
cc @chillee, who noticed that for some benchmarks lowering this number speeds things up. Here you are increasing it.

Reply:
Well, I did also test that in other benchmarks, lowering this number slowed things down by quite a bit :P Basically, this heuristic sucks and we should throw it away ASAP.

Reply:
Do you have an idea what we should replace it with? Why is that vision benchmark different? Currently we have decisions about whether to realize a node or not scattered in many places, so it's hard to see the overall logic.

Reply:
Just to be clear, I renamed the earlier config name from `realize_reads_threadshold` to `realize_reads_remat_threshold` to emphasize that the old flag was trying to avoid exponential rematerialization when you have multiple users. So, this PR does not change the value or behavior of the old flag. It is still set at 4.

This PR adds a new heuristic, named as `realize_reads_threshold`, which keeps a check on how much you could accumulate in a single IR node during lowering. For example, it ensures if you have 1000 add operations one after the other, it does not accumulate all of them in just one IR node. This was the main reason why `hf_T5_large` was taking so much compilation time.

Reply:
Actually let me just keep the old name as it is to avoid confusion. And call new flag something else.